### PR TITLE
DSA - Add support for ksz8863

### PR DESCRIPTION
**Pull request by arvinf**
This PR:
1.  Adds a few bug fixes for the KSZ8794 driver.
2. Refactors the KSZ8794 driver to be more generic for different chips in KSZ8xxx family.
3. Adds support for KSZ8863 chipset
4. Cleans-up the DSA API to pass a switch using the corresponding net interface instead of an integer switch id.
5. Updates the DSA sample to conform to the updated API.

### CODE REVIEW

**Review by mbolivar-nordic — APPROVED** [review]
For the DT binding changes

File: dts/bindings/dsa/microchip_dsa.yaml [hunk 1]
```diff
@@ -0,0 +1,45 @@
+# Copyright (c) 2020 DENX Software Engineering GmbH
+# SPDX-License-Identifier: Apache-2.0
+
+description: |
+    DSA Device
+
+include: [spi-device.yaml]
+properties:
+    dsa-master-port:
```
Comment (cfriedt) [inline dts/bindings/dsa/microchip_dsa.yaml:9]: Suggest rewording 'master' and 'slave' throughout to align with Zephyr's Coding Guidelines
https://docs.zephyrproject.org/latest/contribute/coding_guidelines/index.html#rule-a-2-inclusive-language

**Review by cfriedt — CHANGES_REQUESTED** [review]
Needs some work to align with Coding Guidelines w.r.t. inclusive language.
https://docs.zephyrproject.org/latest/contribute/coding_guidelines/index.html#rule-a-2-inclusive-language

**Comment by jukkar** [comment]
> Needs some work to align with Coding Guidelines w.r.t. inclusive language.

I think the slave/master wording comes from DSA (Distributed Switch Architecture) in Linux which uses these terms https://www.kernel.org/doc/Documentation/networking/dsa/dsa.txt. It would be quite confusing if we start to use different terminology in Zephyr.

File: dts/bindings/dsa/microchip_dsa.yaml (hunk 1 above)
Comment (arvinf) [reply dts/bindings/dsa/microchip_dsa.yaml:9]: I agree w/ @jukkar's comment. The coding guidelines specifically allow exemption for our usecase: "Exceptions are allowed for maintaining existing implementations or adding new implementations of industry standard specifications governed externally to the Zephyr Project.". We can rename the properties when they're changed in Linux.

**Review by cfriedt — APPROVED** [review]
